rewrite-java/src/main/java/org/openrewrite/java/RenamePackage.java

```diff
@@ -0,0 +1,313 @@
+                c = c.withSourcePath(Paths.get(path.replaceFirst(
+                        changingFrom.replace('.', '/'),
+                        changingTo.replace('.', '/')
+                )));
+            }
+            return c;
+        }
+
+        @Override
+        public J.FieldAccess visitFieldAccess(J.FieldAccess fieldAccess, ExecutionContext ctx) {
+            J.FieldAccess f = super.visitFieldAccess(fieldAccess, ctx);
+            if (f.isFullyQualifiedClassReference(oldPackageName)) {
+                if (getCursor().getParent() != null
+                        // Ensure the parent isn't a J.FieldAccess OR the parent doesn't match the target package name.
+                        && (!(getCursor().getParent().getValue() instanceof J.FieldAccess)
+                                || (!(((J.FieldAccess) getCursor().getParent().getValue())
+                                        .isFullyQualifiedClassReference(targetPackageType.getFullyQualifiedName()))))) {
+
+                    f = TypeTree.build(((JavaType.FullyQualified) targetPackageType)
+                            .getFullyQualifiedName()).withPrefix(f.getPrefix());
+                }
+            }
+            return f;
+        }
+
+        @Override
+        public J.Identifier visitIdentifier(J.Identifier ident, ExecutionContext ctx) {
+            // If the J.Identifier's type is equal to the type we're looking for,
+            // and the classname of the type we're looking for is equal to the J.Identifiers string representation
+            // Then transform it, otherwise leave it alone
+            J.Identifier i = super.visitIdentifier(ident, ctx);
+
+            if (TypeUtils.isOfClassType(i.getType(), oldPackageName)
+                    && i.getSimpleName().equals(originalPackageType.getClassName())) {
+
+                i = i.withName((targetPackageType).getClassName())
+                        .withType(targetPackageType);
+            }
+            return i;
+        }
+
+        @Override
+        public J.MethodDeclaration visitMethodDeclaration(J.MethodDeclaration method, ExecutionContext ctx) {
+            J.MethodDeclaration m = super.visitMethodDeclaration(method, ctx);
+            m = m.withReturnTypeExpression(transformName(m.getReturnTypeExpression()));
+            return m.withThrows(m.getThrows() == null ? null : ListUtils.map(m.getThrows(), this::transformName));
+        }
+
+        @Override
+        public J.MethodInvocation visitMethodInvocation(J.MethodInvocation method, ExecutionContext ctx) {
+            J.MethodInvocation m = super.visitMethodInvocation(method, ctx);
+
+            if (m.getSelect() instanceof NameTree
+                    && m.getType() != null && m.getType().hasFlags(Flag.Static)) {
+                m = m.withSelect(transformName(m.getSelect()));
+            }
+
+            if (m.getSelect() != null) {
+                JavaType.Class selectType = TypeUtils.asClass(m.getSelect().getType());
```

Comment:
consider using JavaType.FullyQualified since `TypeUtils.asClass(..)` may return null if the type is an instance of `Cyclic`
```suggestion
                JavaType.FullyQualified selectType = TypeUtils.asFullyQualified(m.getSelect().getType());
```